src/mame/drivers/didact.cpp

```diff
@@ -457,34 +451,381 @@ void mp68a_state::machine_start()
+	PORT_BIT(0x10, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("Q")  PORT_CODE(KEYCODE_Q)  PORT_CHAR('Q')
+	PORT_BIT(0x20, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("1")  PORT_CODE(KEYCODE_1)  PORT_CHAR('1')
+	PORT_BIT(0x40, IP_ACTIVE_LOW,	IPT_UNUSED)
+	PORT_BIT(0x80, IP_ACTIVE_LOW,	IPT_UNUSED)
+
+	PORT_START("LINE2")
+	PORT_BIT(0x01, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("+")  PORT_CODE(KEYCODE_PLUS_PAD)  PORT_CHAR('+')
+	PORT_BIT(0x02, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("-")  PORT_CODE(KEYCODE_MINUS_PAD) PORT_CHAR('-')
+	PORT_BIT(0x04, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("X")  PORT_CODE(KEYCODE_X)  PORT_CHAR('X')
+	PORT_BIT(0x08, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("S")  PORT_CODE(KEYCODE_S)  PORT_CHAR('S')
+	PORT_BIT(0x10, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("W")  PORT_CODE(KEYCODE_W)  PORT_CHAR('W')
+	PORT_BIT(0x20, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("2")  PORT_CODE(KEYCODE_2)  PORT_CHAR('2')
+	PORT_BIT(0x40, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("/")  PORT_CODE(KEYCODE_SLASH_PAD)  PORT_CHAR('/')
+	PORT_BIT(0x80, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("*")  PORT_CODE(KEYCODE_ASTERISK)	PORT_CHAR('*')
+
+	PORT_START("LINE3")
+	PORT_BIT(0x01, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("OUML") PORT_CODE(KEYCODE_COLON) PORT_CHAR(0x00F6) PORT_CHAR(0x00D6)
+	PORT_BIT(0x02, IP_ACTIVE_LOW,	IPT_UNUSED)
+	PORT_BIT(0x04, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME(".")  PORT_CODE(KEYCODE_STOP) PORT_CHAR('.') PORT_CHAR(':')
+	PORT_BIT(0x08, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("L")  PORT_CODE(KEYCODE_L)  PORT_CHAR('L')
+	PORT_BIT(0x10, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("O")  PORT_CODE(KEYCODE_O)  PORT_CHAR('O')
+	PORT_BIT(0x20, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("9")  PORT_CODE(KEYCODE_9)  PORT_CHAR('9')
+	PORT_BIT(0x40, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("P")  PORT_CODE(KEYCODE_P)  PORT_CHAR('P')
+	PORT_BIT(0x80, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("0")  PORT_CODE(KEYCODE_0)  PORT_CHAR('0')
+
+	PORT_START("LINE4")
+	PORT_BIT(0x01, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("AUML") PORT_CODE(KEYCODE_QUOTE) PORT_CHAR('�') PORT_CHAR('�')
+	PORT_BIT(0x02, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("-")  PORT_CODE(KEYCODE_SLASH)  PORT_CHAR('-') PORT_CHAR('_')
+	PORT_BIT(0x04, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME(",")  PORT_CODE(KEYCODE_COMMA)  PORT_CHAR(',') PORT_CHAR(';')
+	PORT_BIT(0x08, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("K")  PORT_CODE(KEYCODE_K)  PORT_CHAR('K')
+	PORT_BIT(0x10, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("I")  PORT_CODE(KEYCODE_I)  PORT_CHAR('I')
+	PORT_BIT(0x20, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("8")  PORT_CODE(KEYCODE_8)  PORT_CHAR('8')
+	PORT_BIT(0x40, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("ARING") PORT_CODE(KEYCODE_OPENBRACE) PORT_CHAR(0x00F7) PORT_CHAR(0x00D7)
+	PORT_BIT(0x80, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("+")  PORT_CODE(KEYCODE_MINUS)  PORT_CHAR('+') PORT_CHAR('?')
+
+	PORT_START("LINE5")
+	PORT_BIT(0x01, IP_ACTIVE_LOW,	IPT_KEYBOARD) PORT_NAME("'/*") /* No good mapping */
```

Comment:
if this is the "extra key" on ISO keyboards next to left shift, it should be `KEYCODE_BACKSLASH2`